vmm/src/device_manager.rs

```diff
@@ -855,6 +855,9 @@ pub struct DeviceManager {
     // Hashmap of PCI b/d/f to their corresponding Arc<Mutex<dyn PciDevice>>.
     pci_devices: HashMap<u32, Arc<dyn Any + Send + Sync>>,
 
+    // BTreeMap of PCI b/d/f to their allocated IRQ.
+    pci_device_irqs: BTreeMap<u32, u32>,
+
```

Comment:
There are now 5 PCI related data structures in this struct. I feel like that points to cleaning this up.

Reply:
I felt the same, but this is kinda different because the other ones will hold only the devices that are present. The point here is to create a list of 32 devices no matter if they're plugged or not, so that INTx would still be supported with VFIO hotplug.
Do you have a better idea on how we could make this work with current structs?

Reply:
Something like a `PCIManager` ? or Move it into `PciBus`

Reply:
Ok let me try something there :)

Reply:
I'm not expecting it to addressed in this PR; but I think more generally `DeviceManager` as a struct and as file is too big and too complex with some very tricky state mechanics.

Reply:
Ok then I'll open an issue so that we can look at how we could simplify `DeviceManager` regarding PCI devices.

Reply:
https://github.com/cloud-hypervisor/cloud-hypervisor/issues/2256